From 1350a8befdb845df81a6a32ed4c9d1beebf083b8 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 01:47:55 -0400
Subject: [iommu] vt-d: Honour intel_iommu=sp_off for non-VMM domains

Message-id: <20140905014755.31297.19031.stgit@gir.stowe>
Patchwork-id: 91314
O-Subject: [RHEL7.1 PATCH 21/65] iommu/vt-d: Honour intel_iommu=sp_off for non-VMM domains
Bugzilla: 1129880 1087643
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1087643
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1129880
Upstream Status: 214e39aa36c9c02355d388f20d83b93fc2fa7298

commit 214e39aa36c9c02355d388f20d83b93fc2fa7298
Author: David Woodhouse <David.Woodhouse@intel.com>
Date:   Wed Mar 19 10:38:49 2014 +0000

    iommu/vt-d: Honour intel_iommu=sp_off for non-VMM domains

    Signed-off-by: David Woodhouse <David.Woodhouse@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index f5934fc..c3d4bc9 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -1624,7 +1624,11 @@ static int domain_init(struct dmar_domain *domain, int guest_width)
  else
   domain->iommu_snooping = 0;
 
- domain->iommu_superpage = fls(cap_super_page_val(iommu->cap));
+ if (intel_iommu_superpage)
+  domain->iommu_superpage = fls(cap_super_page_val(iommu->cap));
+ else
+  domain->iommu_superpage = 0;
+
  domain->nid = iommu->node;
 
  /* always allocate the top pgd */
-- 
1.7.1